server/datastore/mysql/hosts_test.go

```diff
@@ -4360,3 +4362,83 @@ func testHostsReplaceHostBatteries(t *testing.T, ds *Datastore) {
 	require.NoError(t, err)
 	require.ElementsMatch(t, h2Bat, bat2)
 }
+
+func testCountHostsNotResponding(t *testing.T, ds *Datastore) {
+	ctx := context.Background()
+	config := config.FleetConfig{Osquery: config.OsqueryConfig{DetailUpdateInterval: 1 * time.Hour}}
+
+	// responsive
+	_, err := ds.NewHost(ctx, &fleet.Host{
+		OsqueryHostID:       "1",
+		NodeKey:             "1",
+		Platform:            "linux",
+		Hostname:            "host1",
+		DistributedInterval: 10,
+		DetailUpdatedAt:     time.Now().Add(-1 * time.Hour),
+		LabelUpdatedAt:      time.Now(),
+		PolicyUpdatedAt:     time.Now(),
+		SeenTime:            time.Now(),
+	})
+	require.NoError(t, err)
+
+	// not responsive
+	_, err = ds.NewHost(ctx, &fleet.Host{
+		ID:                  2,
+		OsqueryHostID:       "2",
+		NodeKey:             "2",
+		Platform:            "linux",
+		Hostname:            "host2",
+		DistributedInterval: 10,
+		DetailUpdatedAt:     time.Now().Add(-3 * time.Hour),
+		LabelUpdatedAt:      time.Now().Add(-3 * time.Hour),
+		PolicyUpdatedAt:     time.Now().Add(-3 * time.Hour),
+		SeenTime:            time.Now(),
+	})
+	require.NoError(t, err)
+
+	// responsive
+	_, err = ds.NewHost(ctx, &fleet.Host{
+		OsqueryHostID:       "3",
+		NodeKey:             "3",
+		Platform:            "linux",
+		Hostname:            "host3",
+		DistributedInterval: 10,
+		DetailUpdatedAt:     time.Now().Add(-49 * time.Hour),
+		LabelUpdatedAt:      time.Now().Add(-48 * time.Hour),
+		PolicyUpdatedAt:     time.Now().Add(-48 * time.Hour),
+		SeenTime:            time.Now().Add(-48 * time.Hour),
+	})
+	require.NoError(t, err)
+
+	// not responsive
+	_, err = ds.NewHost(ctx, &fleet.Host{
+		OsqueryHostID:       "4",
+		NodeKey:             "4",
+		Platform:            "linux",
+		Hostname:            "host4",
+		DistributedInterval: 10,
+		DetailUpdatedAt:     time.Now().Add(-51 * time.Hour),
+		LabelUpdatedAt:      time.Now().Add(-48 * time.Hour),
+		PolicyUpdatedAt:     time.Now().Add(-48 * time.Hour),
+		SeenTime:            time.Now().Add(-48 * time.Hour),
+	})
+	require.NoError(t, err)
+
+	// was responsive but hasn't been seen in past 7 days so it is not counted
+	_, err = ds.NewHost(ctx, &fleet.Host{
+		OsqueryHostID:       "5",
+		NodeKey:             "5",
+		Platform:            "linux",
+		Hostname:            "host5",
+		DistributedInterval: 10,
+		DetailUpdatedAt:     time.Now().Add(-8 * 24 * time.Hour).Add(-1 * time.Hour),
+		LabelUpdatedAt:      time.Now().Add(-8 * 24 * time.Hour),
+		PolicyUpdatedAt:     time.Now().Add(-8 * 24 * time.Hour),
+		SeenTime:            time.Now().Add(-8 * 24 * time.Hour),
+	})
+	require.NoError(t, err)
+
+	count, err := countHostsNotRespondingDB(ctx, ds.writer, config)
```

Comment:
Let's call this method after each `NewHost` call, so that we know for sure which entries are causing a +1 count.